sqlglot/dialects/mysql.py

```diff
@@ -553,6 +553,26 @@ class Generator(generator.Generator):
             exp.DataType.Type.VARCHAR: "CHAR",
         }
 
+        def limit_sql(self, expression: exp.Limit, top: bool = False) -> str:
+            if not isinstance(expression.expression, exp.Literal):
```

Comment:
maybe we can make this a helper and it can also be used in postgres?

Reply:
postgres doesn't suffer from this problem, but maybe I could just simplify in the base generator instead of overriding so that all dialects can leverage it?

Reply:
postgres also copies an expression and calls simplify

Reply:
Ah that's what you meant, I thought you were talking about simplifying the limit expression. Sure, yea.